src/http/server/handlers/deflate_handler.cr

```diff
@@ -0,0 +1,33 @@
+require "zlib"
+
+class HTTP::DeflateHandler < HTTP::Handler
+  DEFLATE_TYPES = %w(text/html text/plain text/xml text/css text/javascript application/javascript)
```

Comment:
Similar to `Rack::Deflater`: https://github.com/rack/rack/blob/master/lib/rack/deflater.rb#L26